Utilities/bin_patch.cpp

```diff
@@ -705,52 +705,71 @@ std::basic_string<u32> patch_engine::apply(const std::string& name, u8* dst, u32
 				{
 					if (is_all_versions)
 					{
-						patches_for_all_serials_and_all_versions.push_back(patch);
+						patches_for_all_serials_and_all_versions.emplace_back(&patch);
 					}
 					else
 					{
-						patches_for_all_serials_and_this_version.push_back(patch);
+						patches_for_all_serials_and_this_version.emplace_back(&patch);
 					}
 				}
 				else if (is_all_versions)
 				{
-					patches_for_this_serial_and_all_versions.push_back(patch);
+					patches_for_this_serial_and_all_versions.emplace_back(&patch);
 				}
 				else
 				{
-					patches_for_this_serial_and_this_version.push_back(patch);
+					patches_for_this_serial_and_this_version.emplace_back(&patch);
 				}
 
 				break;
 			}
 		}
 	}
 
-	// Sort specific patches in front of global patches
-	std::vector<patch_engine::patch_info> sorted_patches;
-	sorted_patches.insert(sorted_patches.end(), patches_for_this_serial_and_this_version.begin(), patches_for_this_serial_and_this_version.end());
-	sorted_patches.insert(sorted_patches.end(), patches_for_this_serial_and_all_versions.begin(), patches_for_this_serial_and_all_versions.end());
-	sorted_patches.insert(sorted_patches.end(), patches_for_all_serials_and_this_version.begin(), patches_for_all_serials_and_this_version.end());
-	sorted_patches.insert(sorted_patches.end(), patches_for_all_serials_and_all_versions.begin(), patches_for_all_serials_and_all_versions.end());
-
 	// Apply modifications sequentially
-	for (const auto& patch : sorted_patches)
+	auto apply_func = [&](const patch_info& patch)
 	{
-		if (!patch.patch_group.empty())
-		{
-			if (m_applied_groups.contains(patch.patch_group))
-			{
-				continue;
-			}
-
-			m_applied_groups.insert(patch.patch_group);
-		}
-
 		auto applied = apply_modification(patch, dst, filesz, min_addr);
 
 		applied_total += applied;
 
 		patch_log.success("Applied patch (hash='%s', description='%s', author='%s', patch_version='%s', file_version='%s') (<- %u)", patch.hash, patch.description, patch.author, patch.patch_version, patch.version, applied.size());
+	};
+
+	const auto patch_super_list =
+	{
+		&patches_for_all_serials_and_all_versions,
+		&patches_for_all_serials_and_this_version,
+		&patches_for_this_serial_and_all_versions,
+		&patches_for_this_serial_and_this_version
+	};
+
+	// Filter by patch group
+	for (auto it = std::rbegin(patch_super_list); it != std::rend(patch_super_list); it++)
```

Comment:
This still looks like you add global group patches first and then skip specific patches if they are in the same group.

Reply:
No it checks spefici patches first.

Reply:
Ah sorry. I didn't see the rbegin